update smart sensor docs and minor fix on is_smart_sensor_compatible() (#22386)
mistercrunch/Airflow: docs/apache-airflow/concepts/smart-sensors.rst

enabling/disabling the smart sensor. Rotating centralized smart sensor tasks will not
cause any user’s sensor task failure.

> *   Callback arguments(``on_success_callback``, ``on_failure_callback``, and ``on_retry_callback``) on sensor operator are not compatible with smart sensor. If any callback arguments is provided, that sensor task will not be executed on smart sensor.

Reviewer: ```suggestion
*   Using callback arguments (``on_success_callback``, ``on_failure_callback``, and ``on_retry_callback``) on a sensor task is not compatible with the smart sensor mode. If any callback arguments are provided, the sensor task will not be executed when the smart sensor mode is enabled.
```
